docs/concepts.md

```diff
@@ -1,27 +1,213 @@
+| (3) *Kubernetes [namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/)* | dev | Pulled from current [kubeconfig context](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#context) OR specified at command line | Inside `environments/<env-name>/spec.json` |
+| (4) *Kubernetes API version* | version:v1.7.1 | Specified at command line, otherwise defaults to v1.7.0 | Used to generate appropriate files in `environments/<env-name>/.metadata/` based on the specified version of the Kubernetes OpenAPI spec |
+
+ksonnet allows you to deploy any particular *application* to **multiple** environments. Below is a visualization of two environments that represent different namespaces on the same cluster:
+
+![ksonnet environment diagram](/docs/img/environment.svg)
+
+If you're wondering why you might deploy a common set of manifests to different environments, here are some potential use cases:
+* Release Management (*dev* vs *test* vs *prod*)
+* Multi-AZ (*us-west-2* vs *us-east-1*)
+* Multi-cloud (*AWS* vs *GCP* vs *Azure*)
+
+---
 
 ### Component
-TODO
+
+A ksonnet *application* can be broken down into a series of discrete components:
+
+<p align="center">
+<img alt="ksonnet component diagram" src="/docs/img/component.svg" height="150px">
+</p>
+
+Components can be as simple as a single Kubernetes resource (e.g. a Deployment) or as complex as a complete logging stack (e.g. EFK). More concretely, a component corresponds to a Kubernetes manifest in `components/`. This manifest is autogenerated by the [`ks generate`](docs/cli-reference/ks_generate.md) command, and expressed in a language called [Jsonnet](#jsonnet).
+
+How are these components generated? All components are generated from *prototypes*. The distinction between a component and a prototype is a bit subtle. If you are familiar with object oriented programming, a prototype is like a class, and a component is like its instantiation:
+
+<p align="center">
+<img alt="component class analogy" src="/docs/img/component_and_prototype.svg" height="180px">
+</p>
+
+All of the component files in an *app* can be deployed to a specified *environment* using [`ks apply`](docs/cli-reference/ks_apply.md).
+
+---
 
 ### Prototype
-TODO
+
+Prototypes are the basis for components—they are the reason that `ks generate` allows you to quickly move from a blank app to multiple complete manifests.
+
+By itself, a prototype is an pre-written but incomplete manifest:
+
+<p align="center">
+<img alt="prototype diagram" src="/docs/img/prototype.svg" height="170px">
+</p>
+
+Why is this useful? Prototypes allow you to avoid copying and pasting boilerplate code, and to focus on the parts of your configuration that are specific to your app. Specifically, during `ks generate`, you can specify certain command-line *parameters* to "fill-in-the-blank(s)" of a prototype and output a complete `component/` manifest:
```

Comment:
I wrote a comment here, and lost it. I'll restate it succinctly.

- [ ] Write "fill-in-the-blanks" pluralized, as it's sufficiently unlikely that a prototype will fewer than two parameters.
- [ ] Is the trailing slash after "componet/" intentional?